Catappult/appcoins-wallet-android
# [feature/APPC-3403] Pending purchase info from the backend

**What does this PR do?**

   Adds the pending cached purchase from the backend to be able to trigger the Onboarding screen from every source and not only from the Google play.
   Keeps the Google play check as a fallback

**Database changed?**

   No

**Where should the reviewer start?**

- AppStart.kt
- PendingPurchaseFlowUseCase.kt

**How should this be manually tested?**

  - Uninstall the wallet and start a payment, to trigger the web view
  - Install the wallet, can be from the Google Play, Aptoide, or any other source
  - Check the Onboarding from the Game screen shows 

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/APPC-3403

**Code Review Checklist**
